src/main/kotlin/org/jitsi/jicofo/jibri/BaseJibri.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015-Present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.jibri
+
+import org.jitsi.impl.protocol.xmpp.ChatRoomMemberRole
+import org.jitsi.impl.protocol.xmpp.XmppProvider
+import org.jitsi.jicofo.JitsiMeetConferenceImpl
+import org.jitsi.jicofo.TaskPools
+import org.jitsi.jicofo.jibri.JibriSession.StateListener
+import org.jitsi.jicofo.util.ErrorResponse.create as error
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.queue.PacketQueue
+import org.jitsi.xmpp.extensions.jibri.JibriIq
+import org.jitsi.xmpp.extensions.jibri.JibriIq.Action
+import org.jivesoftware.smack.AbstractXMPPConnection
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.XMPPError
+import org.jivesoftware.smack.SmackException
+
+/**
+ * Base class shared between [JibriRecorder] (which can deal with only one [JibriSession]) and [JibriSipGateway]
+ * (which is capable of handling multiple, simultaneous [JibriSession]s).
+ *
+ * @author Pawel Domas
+ */
+abstract class BaseJibri internal constructor(
+    protected val conference: JitsiMeetConferenceImpl,
+    private val xmppProvider: XmppProvider,
+    parentLogger: Logger,
+    val jibriDetector: JibriDetector
+) : StateListener {
+
+    protected val connection: AbstractXMPPConnection = xmppProvider.xmppConnection
+
+    private val incomingIqQueue = PacketQueue<JibriIq>(
+        50,
+        true,
+        "jibri-iq-queue-${conference.roomName.localpart}",
+        { jibriIq ->
+            val response = doHandleIQRequest(jibriIq)
+            try {
+                xmppProvider.xmppConnection.sendStanza(response)
+            } catch (e: SmackException.NotConnectedException) {
+                logger.warn("Failed to send response, smack is not connected.")
+            } catch (e: InterruptedException) {
+                logger.warn("Failed to send response, interrupted.")
+            }
+
+            true
+        },
+        TaskPools.ioPool
+    )
+
+    /**
+     * The logger instance pass to the constructor that wil be used by this
+     * instance for logging.
+     */
+    protected val logger: Logger = parentLogger.createChildLogger(BaseJibri::class.simpleName)
```

Comment:
Now that we're using `Logger2` do we want to add conference-specific parameters here?

Reply:
I don't know that we can add anything useful, it already inherits the conference name.